File: YSY/11_25/코드 처리하기.py
def solution(code):
    answer = ''
    ret=[]
    mode=False
    for idx, letter in enumerate(code):
        if not mode:
            if (letter!="1") and (idx%2==0):
                ret.append(letter)
            elif letter=="1":
                mode=True
        else:
            if (letter!="1") and (idx%2==1):
                ret.append(letter)
            elif letter=="1":
                mode=False
    answer=''.join(ret)
    
    if not answer:
        answer="EMPTY"
    return answer

File: JJB/231126/코드_처리하기.cpp
#include <string>
#include <vector>
#include <numeric>

using namespace std;

string solution(string code) 
{
    string answer = "";
    bool mode = false;
    
    for(int idx = 0; idx < code.size(); idx++)
    {
        if(code[idx] == '1')
            mode = !mode;
        else if(!mode && idx%2==0)
            answer.push_back(code.at(idx));
        else if(mode && idx%2==1)
            answer.push_back(code.at(idx));
    }
    
    if(answer.empty())
        answer = "EMPTY";
    
    return answer;
}